platform/ios/src/MGLScaleBarView.m

```diff
@@ -0,0 +1,289 @@
+#import <Mapbox/Mapbox.h>
+
+#import "MGLScaleBarView_Private.h"
+
+@interface MGLScaleBarView()
+@property (nonatomic, assign, getter=isVisible) BOOL visible;
```

Comment:
`hidden` would hide the view instantly so I use this property to flag the view as visible/invisible when fadeIn/fadeOut begins in order to prevent unwanted calls to those methods.